integration-tests/src/test/java/oracle/kubernetes/operator/ItModelInImageOverride.java

```diff
@@ -0,0 +1,246 @@
+  private static String domainNS;
+  private static String testClassName;
+  private static StringBuffer namespaceList;
+  private static final String configMapSuffix = "-mii-config-map";
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * @throws Exception exception
+   */
+  @BeforeAll
+  public static void staticPrepare() throws Exception {
+    namespaceList = new StringBuffer();
+    testClassName = new Object() {
+    }.getClass().getEnclosingClass().getSimpleName();
+    // initialize test properties and create the directories
+    initialize(APP_PROPS_FILE, testClassName);
+  }
+
+  /**
+   * This method gets called before every test. It creates the result/pv root directories
+   * for the test. Creates the operator and domain if its not running.
+   *
+   * @throws Exception exception if result/pv/operator/domain creation fails
+   */
+  @BeforeEach
+  public void prepare() throws Exception {
+    createResultAndPvDirs(testClassName);
+
+    // create operator1
+    if (operator == null) {
+      Map<String, Object> operatorMap = createOperatorMap(getNewSuffixCount(),
+          true, testClassName);
+      operator = TestUtils.createOperator(operatorMap, RestCertType.SELF_SIGNED);
+      Assertions.assertNotNull(operator);
+      domainNS = ((ArrayList<String>) operatorMap.get("domainNamespaces")).get(0);
+      namespaceList.append((String)operatorMap.get("namespace"));
+      namespaceList.append(" ").append(domainNS);
+    }
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories.
+   *
+   * @throws Exception exception
+   */
+  @AfterAll
+  public static void staticUnPrepare() throws Exception {
+    tearDown(new Object() {
+    }.getClass().getEnclosingClass().getSimpleName(), namespaceList.toString());
+
+    LoggerHelper.getLocal().info("SUCCESS");
+  }
+
+  /**
+   * Create a domain using model in image and having configmap in the domain.yaml
+   * before deploying the domain. After deploying the domain crd,
+   * re-create the configmap with a model file that define a JDBC DataSource
+   * and update the domain crd to new config map and change domain
+   * restartVersion to reload the model, generate new config and initiate a
+   * rolling restart.
+   *
+   * @throws Exception exception
+   */
+  @Test
+  public void testMiiOverrideNonExistJdbc() throws Exception {
+    Assumptions.assumeTrue(QUICKTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+    LoggerHelper.getLocal().log(Level.INFO,
+        "Creating Domain & waiting for the script to complete execution");
+    boolean testCompletedSuccessfully = false;
+    try {
+      // create Domain using the image created by MII
```

Comment:
this is a really nice test, i like this, i can read it understand the intent easily.

Reply:
Thanks. I just added a python file to verify dsResource values like readTimeout, DS name, etc. Ready for the second test case :)